Libs/Optimize/ParticleSystem/itkParticleGradientDescentPositionOptimizer.txx

```diff
@@ -162,10 +162,36 @@ namespace itk
               VectorType gradient = original_gradient_projectedOntoTangentSpace * m_TimeSteps[dom][k];
 
               // Step B Constrain the gradient so that the resulting position will not violate any domain constraints
-              m_ParticleSystem->GetDomain(dom)->GetConstraints()->applyBoundaryConstraints(gradient, m_ParticleSystem->GetPosition(k, dom));
+              //m_ParticleSystem->GetDomain(dom)->GetConstraints()->applyBoundaryConstraints(gradient, m_ParticleSystem->GetPosition(k, dom));
 
               gradmag = gradient.magnitude();
 
+              // Step B 1: if the magnitude is larger than the Sampler allows, scale the gradient down to an acceptable magnitude
```

Comment:
This seems like a lot of lines to add to the main body of the gradient descent optimizer, can this new section be placed in a separate function?  Also, should there be a flag that enables or disables this, or does this all need to be run even with no constraints?